### DISCUSSION TITLE
dvc exp save target

### DISCUSSION THREAD

**Discussion by gregstarr** [treeverse/dvc#10236] [General]
How come we can `dvc exp run <target>` but we can't `dvc exp save <target>`?

**Comment by pmrowla** [comment]
Experiment commits capture the state of your entire repository at the time of the commit (and not just the state of a specific target).

So for `dvc exp save` specifying a target wouldn't actually have any effect (it just captures the repository state at the time you run the command). For `dvc exp run`, you can specify a target because DVC reproduces those stage(s) before capturing the repository state.

**Reply by gregstarr** [reply]
When you `dvc exp run <target>` does DVC capture the state of the entire repository or just the target?

**Reply by pmrowla** [reply]
It captures the state of the entire repository

**Reply by gregstarr** [reply]
Do `dvc exp run` and `dvc exp save` have different behavior when it comes to saving though? `dvc exp save` complains if things are missing e.g.
```
ERROR: failed to save experiment - output 'pipelines/big/models/surface.keras' does not exist 
```
but `dvc exp run <target>` doesn't have any problem with this.

**Reply by pmrowla** [reply]
This is probably an `exp save` bug, it should behave the same as `exp run` in this situation (and just save whatever is available). Feel free to open a separate issue for this

**Reply by dberenbaum** [reply]
Does `dvc exp run` generate that `pipelines/big/models/surface.keras` output? I think both will fail if that does not exist by the end of the experiment.

**Reply by gregstarr** [reply]
no `exp run` will not generate that output. `exp run` will only sort-of fail. It will still reproduce the outputs I want and save them. The experiment will be red in the vscode extension and complain about something failing but it will still basically work.

**Reply by dberenbaum** [reply]
Got it. So if you have two pipelines, you can run one and save the results with `dvc exp run <target>` even if the other is missing outputs.

@pmrowla I think this would also fail in `dvc exp run` without a target. Consider this `dvc.yaml`:


```yaml
stages:
  one:
    cmd: echo one > one.txt
    outs:
      - one.txt
  two:
    cmd: echo two > /dev/null
    outs:
      - two.txt
```

`dvc exp run` fails with the error `ERROR: failed to reproduce 'two': output 'two.txt' does not exist`. I think it makes sense to support `dvc exp save <target>`, which will narrow the `dvc commit` part of the operation (even though the `git commit` is for the entire repo).
